github/handlers/ExecuteBlockActionHandler.ts

```diff
@@ -735,11 +735,18 @@ export class ExecuteBlockActionHandler {
                     let value: string = context.getInteractionData().value as string;
                     let splittedValues = value?.split(" ");
                     let { user } = await context.getInteractionData();
+                    let accessToken = await getAccessTokenForUser(this.read, user, this.app.oauth2Config) as IAuthData;
+                    this.app.getLogger().debug(accessToken.token)
                     if(splittedValues.length==2){
                         let repoName = splittedValues[0];
                         let issueNumber = splittedValues[1];
-                        let issueComments = await getIssuesComments(this.http,repoName,issueNumber);
-                        let issueData = await getIssueData(repoName,issueNumber,' ',this.http);
+                        if(accessToken){
+                            var issueComments = await getIssuesComments(this.http,repoName,accessToken.token,issueNumber);
+                            var issueData = await getIssueData(repoName,issueNumber,accessToken.token,this.http)    
+                        }else{
+                            var issueComments = await getIssuesComments(this.http,repoName,null,issueNumber);
+                            var issueData = await getIssueData(repoName,issueNumber,null,this.http)
+                        }
```

Comment:
I think we can do something like this and github sdk methods will handle  the rest

```

let issueComments = await getIssuesComments(this.http,repoName,accessToken?.token,issueNumber);
let issueData = await getIssueData(repoName,issueNumber,accessToken?.token,this.http) ;

```